Sources/ODP/ZaiusGraphQLApiManager.swift

```diff
@@ -0,0 +1,238 @@
+ curl -i -H 'Content-Type: application/json' -H 'x-api-key: W4WzcEs-ABgXorzY7h1LCQ' -X POST -d '{"query":"query {customer(fs_user_id: \"tester-101\") {audiences(subset:[\"has_email\",\"has_email_opted_in\",\"push_on_sale\"]) {edges {node {name state}}}}}"}' https://api.zaius.com/v3/graphql
+
+ // fetch info with vuid for ["has_email", "has_email_opted_in", "push_on_sale"] segments
+ curl -i -H 'Content-Type: application/json' -H 'x-api-key: W4WzcEs-ABgXorzY7h1LCQ' -X POST -d '{"query":"query {customer(vuid: \"d66a9d81923d4d2f99d8f64338976322\") {audiences(subset:[\"has_email\",\"has_email_opted_in\",\"push_on_sale\"]) {edges {node {name state}}}}}"}' https://api.zaius.com/v3/graphql
+
+ query MyQuery {
+   customer(vuid: "d66a9d81923d4d2f99d8f64338976322") {
+     audiences(subset:["has_email","has_email_opted_in","push_on_sale"]) {
+       edges {
+         node {
+           name
+           state
+         }
+       }
+     }
+   }
+ }
+
+ [GraphQL Response]
+ 
+ {
+   "data": {
+     "customer": {
+       "audiences": {
+         "edges": [
+           {
+             "node": {
+               "name": "has_email",
+               "state": "qualified",
+             }
+           },
+           {
+             "node": {
+               "name": "has_email_opted_in",
+               "state": "qualified",
+             }
+           },
+            ...
+         ]
+       }
+     }
+   }
+ }
+ 
+ [GraphQL Error Response]
+
+ {
+   "errors": [
+     {
+       "message": "Exception while fetching data (/customer) : java.lang.RuntimeException: could not resolve _fs_user_id = asdsdaddddd",
+       "locations": [
+         {
+           "line": 2,
+           "column": 3
+         }
+       ],
+       "path": [
+         "customer"
+       ],
+       "extensions": {
+         "classification": "InvalidIdentifierException"
+       }
+     }
+   ],
+   "data": {
+     "customer": null
+   }
+ }
+*/
+
+class ZaiusGraphQLApiManager {
+    let logger = OPTLoggerFactory.getLogger()
+
+    func fetchSegments(apiKey: String,
+                       apiHost: String,
+                       userKey: String,
+                       userValue: String,
+                       segmentsToCheck: [String],
+                       completionHandler: @escaping ([String]?, OptimizelyError?) -> Void) {
+        let subsetFilter = makeSubsetFilter(segments: segmentsToCheck)
+        
+        let body = [
+            "query": "query {customer(\(userKey): \"\(userValue)\") {audiences\(subsetFilter) {edges {node {name state}}}}}"
+        ]
+        guard let httpBody = try? JSONEncoder().encode(body) else {
+            completionHandler(nil, .fetchSegmentsFailed("invalid query."))
+            return
+        }
+
+        let apiEndpoint = apiHost + "/v3/graphql"
+        let url = URL(string: apiEndpoint)!
+        var urlRequest = URLRequest(url: url)
+        urlRequest.httpMethod = "POST"
```

Comment:
Can we have this code inside a method to reuse it? this is also there in some other classes.

Reply:
I'd keep it as is since not much gain by reusing these lines.